src/geo/geocoder/mapbox-geocoder.js

```diff
@@ -0,0 +1,84 @@
+var ENDPOINT = 'https://api.mapbox.com/geocoding/v5/mapbox.places/{{address}}.json?access_token={{access_token}}';
+
+var TYPES = {
+  country: 'country',
+  region: 'region',
+  postcode: 'postal-area',
+  district: 'localadmin',
+  place: 'venue',
+  locality: 'locality',
+  neighborhood: 'neighbourhood',
+  address: 'address',
+  poi: 'venue',
+  'poi.landmark': 'venue'
+};
+
+function MapboxGeocoder () { }
+
+MapboxGeocoder.geocode = function (address, token) {
+  if (!address) {
+    throw new Error('MapboxGeocoder.geocode called with no address');
+  }
+  if (!token) {
+    throw new Error('MapboxGeocoder.geocode called with no access_token');
+  }
+  return fetch(ENDPOINT.replace('{{address}}', address).replace('{{access_token}}', token))
+    .then(function (response) {
+      return response.json();
+    })
+    .then(function (response) {
+      return _formatResponse(response);
+    });
+};
+
+/**
+ * Transform a mapbox geocoder response on a object friendly with our search widget.
+ * @param {object} rawMapboxResponse - The raw mapbox geocoding response, {@see https://www.mapbox.com/api-documentation/?language=JavaScript#response-object}
+ */
+function _formatResponse (rawMapboxResponse) {
+  if (!rawMapboxResponse.features.length) {
+    return [];
+  }
+  var center = _getCenter(rawMapboxResponse.features[0]);
+  return [{
+    center: center,
+    lat: center.lat,
```

Comment:
`center` is returned by `_getCenter` that returns an array. But then we fill `lat` and `long` with `center.lat` and `center.long` accessing as if `center` was an object instead of an array.